### PR TITLE
8372 new ui elements for viewers

### LINKED ISSUE #8372: New UI elements for viewers
- [X] Profile page badge.
- [X] Dropdown menu badge.
- [X] Footer message about becoming a builder.
- [X] "Become a Builder" button and link.

cc @urbanphes

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #8372.

@xavijam, @javitonino , could you CR front and back code, please? I'll merge this into #8047 before deploying the code (I will deploy the migrations meanwhile).

### CODE REVIEW

File: lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown.jst.ejs [hunk 1]
```diff
@@ -8,6 +8,19 @@
       <small class="DefaultDescription"><%- email %></small>
     </p>
   </li>
+    <li class="SettingsDropdown-item">
+      <p class="SettingsDropdown-userRole">
+        <% if (isViewer) { %>
+          <span class="UserRoleIndicator Viewer">VIEWER</span>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown.jst.ejs]: For the future, using our new CSS rules, this should be: 
`... class="UserRoleIndicator UserRoleIndicator--viewer" ...`

**Comment by xavijam** [comment]
:+1:

File: lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown.jst.ejs (hunk 1 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown.jst.ejs]: Ok. Can I keep it this time? U_U Main PR also has some of these 👉 👈

**Comment by juanignaciosl** [comment]
@javitonino don't CR this yet, please. I don't know what happened but something has stopped working U_U . I must fix dashboard links behaviour, please wait. Sorry U_U

**Comment by juanignaciosl** [comment]
@javitonino go ahead, please. Backend, bulletproof fix has been easier than expected :-)

**Comment by javitonino** [comment]
LGTM

**Pull request merged**